pkg/scheduler/queue/tenant_queues.go

```diff
@@ -127,84 +122,60 @@ func newQueueBroker(maxTenantQueueSize int, forgetDelay time.Duration) *queueBro
 	}
 }
 
-func (qb *queueBroker) len() int {
-	return len(qb.tenantQueues)
+func (qb *queueBroker) isEmpty() bool {
+	return qb.tenantQueuesTree.IsEmpty()
 }
 
-func (qb *queueBroker) enqueueRequestBack(request *tenantRequest) error {
-	queue, err := qb.getOrAddTenantQueue(request.tenantID, request.maxQueriers)
+// enqueueRequestBack is the standard interface to enqueue requests for dispatch to queriers.
+//
+// Tenants and tenant-querier shuffle sharding relationships are managed internally as needed.
+func (qb *queueBroker) enqueueRequestBack(request *tenantRequest, tenantMaxQueriers int) error {
+	_, err := qb.tenantQuerierAssignments.getOrAddTenant(request.tenantID, tenantMaxQueriers)
```

Comment:
It looks like the tenant returned from `getOrAddTenant` is now not used anywhere this is called - should this return value be dropped? If so, should we also rename the method to something like `createOrUpdateTenant`?

Reply:
yep, I like that